hy/compiler.py

```diff
@@ -2071,14 +2071,35 @@ def rewire_init(expr):
 
         bases_expr = []
         bases = Result()
+        keywords = []
         if expressions:
             base_list = expressions.pop(0)
             if not isinstance(base_list, HyList):
                 raise HyTypeError(base_list, "Base classes must be a list.")
-            bases_expr, bases, _ = self._compile_collect(base_list)
+            bases_expr, bases, keywords = self._compile_collect(base_list, with_kwargs=True)
 
         body = Result()
 
+        if not PY3 and keywords:
+            # Dirty shim for Python 2 to make :metaclass keyword work.
+            if len(keywords) != 1 or keywords[0].arg != 'metaclass':
+                raise HyTypeError(base_list, "python 2 only supports a "
+                                             "metaclass keyword argument")
+
+            if not isinstance(keywords[0].value, ast.Name):
+                raise HyTypeError(base_list, "specified metaclass must be "
+                                             "a symbol")
+
+            e = HyExpression([
+                    HySymbol("setv"),
+                    HySymbol("__metaclass__"),
+                    HySymbol(keywords[0].value.id)
+                ]).replace(base_list)
+            spoof_positions(e)
```

Comment:
Do you need both `.replace` and `spoof_positions` here? Shouldn't one be enough?

Reply:
Yeah, I thought it was weird when I looked at it, `replace` should be enough and idea. But I copy and pasted the approach and elsewhere both are used:

https://github.com/hylang/hy/blob/10cd36d13d71010c38e8b411ebf13390f80b9542/hy/compiler.py#L414-L424

Reply:
Such cases might also be redundant, couldn't they? (Not that I'm expecting you to fix them in this PR.)

Reply:
Yeah, agreed. Just pointing it out so we know. I'm certainly going to investigate it.